### PR TITLE
Add pennylane eco system

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/openfermionpyscf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - openfermion >=0.5
+    - pyscf
+    - pytest
```
Comment (BastianZim) [inline recipes/openfermionpyscf/meta.yaml:25]: Yes, this is really listed as required for some reason.

File: recipes/pennylane-qchem/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{% set name = "pennylane-qchem" %}
+{% set version = "0.17.0" %}
```
Comment (BastianZim) [inline recipes/pennylane-qchem/meta.yaml:2]: I need this version for a different package. Will update it in the feedstock.

**Comment by BastianZim** [comment]
@conda-forge/help-python Would be ready for review, thanks!

Only Linux currently works because of missing dependencies but it's noarch.

**Comment by BastianZim** [comment]
Gentle ping @conda-forge/staged-recipes

File: recipes/openfermionpyscf/meta.yaml (hunk 1 above)
Comment (BastianZim) [reply recipes/openfermionpyscf/meta.yaml:25]: I've added that comment so often when reviewing... :)

Yes but that's not yet the case for 0.5, see here: https://github.com/quantumlib/OpenFermion-PySCF/blob/v0.5/requirements.txt

Will remove that with the next update though!

Comment (carterbox) [reply recipes/openfermionpyscf/meta.yaml:25]: I see! You want `pip check` to pass, so it has to be there.

**Comment by BastianZim** [comment]
Thanks for merging

**Pull request merged**